core/app/models/spree/order_cancellations.rb

```diff
@@ -42,6 +42,36 @@ def short_ship(inventory_units, whodunnit:nil)
     unit_cancels
   end
 
+  def cancel_unit(inventory_unit, reason: Spree::UnitCancel::DEFAULT_REASON, whodunnit:nil)
```

Comment:
:+1: yanking `short_ship_unit` and re-using this method instead seems like a great idea and a good small step for this PR in making `short_ship` a bit better.